### ISSUE TITLE
Clean up PANIC_DUE_TO_CORRUPT_ENUM macros

### ISSUE THREAD

**Issue by alyssawilk** [envoyproxy/envoy#19229]
https://github.com/envoyproxy/envoy/pull/19204 replaced NOT_REACHED_GCOVR_EXCL_LINE macros with more clear PANIC_DUE_TO_CORRUPT_ENUM macros.

While more clear, the panics are still suboptimal becasue doing break; instead of return; results in production crashes.  These should be cleaned up and replaced with sensible defaults on a case by case basis.

**Comment by tommyp1ckles** [comment]
I'd like to work on this.

**Comment by ankatare** [comment]
@alyssawilk its to wait code to be merged for #19204  ?

**Comment by alyssawilk** [comment]
yeah sorry it's taking longer than expected

**Comment by ankatare** [comment]
@alyssawilk Oh Ok. Sure.

**Comment by ankatare** [comment]
@tommyp1ckles whould you still be working on this ?

**Comment by tommyp1ckles** [comment]
@alyssawilk yes. I think I need some clarification though. Are we expecting to replace all panic_due_to_corrupt_enum or just where it makes sense?

**Comment by alyssawilk** [comment]
I think in general we don't want panics in the Envoy code base, especially ones that could be hit by accident.
If it's the control plane, we should throw an error, and if it's the data plane we should have error handling.

**Comment by burnerlee** [comment]
@alyssawilk can I take up this issue, if no-one is working on this currently?

**Comment by tommyp1ckles** [comment]
@burnerlee I'm working on this now, was busy with other things. We could split up the work if you'd like.

**Comment by burnerlee** [comment]
@tommyp1ckles oh I see. Sure, we can split up the work. Thank you so much

**Comment by tommyp1ckles** [comment]
@burnerlee great, I'm currently working on all the uses in common/upstream. I was going to move onto access_log and config next.

**Comment by tommyp1ckles** [comment]
@burnerlee What files where you planning on working through?
